vip-scanner/checks/VIPRestrictedVariablesCheck.php

```diff
@@ -0,0 +1,36 @@
+<?php
+
+class VIPRestrictedVariablesCheck extends BaseCheck
+{
+	function check( $files ) {
+		$result = true;
+
+		$checks = array(
+			'((?<![\\\'\"])\$\$[a-zA-Z_\x7f-\xff][a-zA-Z0-9_\x7f-\xff]*?(?![\\\'\"])|(?<![\\\'\"])\$\{(?:.*)[\}](?![\\\'\"]))' => array( "level" => "Warning", "note" => "Possible variable variables" ),
```

Comment:
>Will this regex match variable class properties?
>`$foo->$bar`

No, good catch.  There's still probably more odd and extreme cases that it's not covering.

`\$[a-zA-Z_\x7f-\xff][a-zA-Z0-9_\x7f-\xff]*?\-\>\$` should match this case.

The full regex would end up being:
`((?<![\'\"])\$\$[a-zA-Z_\x7f-\xff][a-zA-Z0-9_\x7f-\xff]*?(?![\'\"])|(?<![\'\"])\$[a-zA-Z_\x7f-\xff][a-zA-Z0-9_\x7f-\xff]*?\-\>\$(?![\'\"])|(?<![\'\"])\$\{(?:.*)[\}](?![\'\"]))`